docs/design_spec.md

```diff
@@ -0,0 +1,94 @@
+# Nimbess Agent Design
+
+## Overview
+
+Nimbess Agent's main purpose is to manage an external data plane. The Agent's main responsibilities
+include:
+
+ 1. Translating Northbound (NB) application-specific resources into network oriented resources across the
+ entire network fabric.
+ 2. Managing and serving as the control plane for the locally assigned, and external data plane (Southbound).
+ 
+ The Agent is distributed over every host and each Agent instance is capable of viewing and managing the
+ network configuration for the entire fabric. However, only the Agent local to a data plane (on the same host)
+ is capable of directly configuring that data plane.
+ 
+## Agent Northbound Design
+ 
+Nimbess Agent is able to support multiple container platforms and by default supports Kubernetes (K8S). It
```

Comment:
Should we elaborate on "able" - meaning we support only K8S today, but the architecture could be extended to support more. Or do you think that is there today and no changes would be needed in the agent to support something else?

Reply:
Well it would be able to support any platform which implements CNI. For supporting platform specific components like k8s services, etc we would need to add those equivalent renderers. The southbound logic should not need to change to support this.